ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -604,6 +628,37 @@ func (r *RayClusterReconciler) updateStatus(instance *rayiov1alpha1.RayCluster)
 	// TODO (@Jeffwan): Update state field later.
 	// We always update instance no matter if there's one change or not.
 	instance.Status.LastUpdateTime.Time = time.Now()
+
+	// validation for the RayStartParam for the state.
+	isValid, err := common.ValidateHeadRayStartParams(instance.Spec.HeadGroupSpec)
+	if err != nil {
+		r.Recorder.Event(instance, v1.EventTypeWarning, "Parameters conflict", err.Error())
+	}
+	// only in invalid status that we update the status to unhealthy.
+	if !isValid {
+		instance.Status.State = rayiov1alpha1.UnHealthy
+	} else {
+		totalDesireReplicas := *instance.Spec.HeadGroupSpec.Replicas + utils.CalculateDesiredReplicas(instance)
+		if utils.CalculateAvailableReplicas(runtimePods) >= totalDesireReplicas && utils.CheckAllPodsRunnning(runtimePods) {
+			instance.Status.State = rayiov1alpha1.Ready
+		}
+	}
+
+	if len(runtimeServices.Items) != 0 {
+		svc := runtimeServices.Items[0]
+		if instance.Status.Endpoints == nil {
+			instance.Status.Endpoints = map[string]int32{}
+		}
+		for _, port := range svc.Spec.Ports {
+			if len(port.Name) == 0 {
+				r.Log.Info("updateStatus", "service port name is empty", port)
+				continue
+			}
+			// TODO: Endpoints may be misleading for port
```

Comment:
In the previous discuss, I think its may represent the endpoints' port, not the real endpoints? we must combine it with the address so I think it may need a new name here